Sources/NIO/EventLoop.swift

```diff
@@ -608,6 +608,35 @@ extension EventLoop {
         }
     }
 
+    /// Creates and returns a new `EventLoopFuture` that is marked as succeeded or failed with the value held by `result`.
+    ///
+    /// - Parameters:
+    ///   - result: The value that is used by the `EventLoopFuture`
+    /// - Returns: A completed `EventLoopFuture`.
+    @inlinable
+    public func makeCompletedFuture<Success>(_ result: Result<Success, Error>) -> EventLoopFuture<Success> {
+        switch result {
+        case .success(let value):
+            return self.makeSucceededFuture(value)
+        case .failure(let error):
+            return self.makeFailedFuture(error)
+        }
+    }
+
+    /// Creates and returns a new `EventLoopFuture` that is marked as succeeded or failed with the value held by `result`.
+    ///
+    /// - Parameters:
+    ///   - result: The value that is used by the `EventLoopFuture`
+    /// - Returns: A completed `EventLoopFuture`.
+    public func makeCompletedVoidFuture(_ result: Result<Void, Error>) -> EventLoopFuture<Void> {
```

Comment:
So technically, we don't need these two methods. We can just use the fully generic one and do:

```
if Success.self == Void.self {
    // code for makeCompletedVoidFuture here
} else {
   // code for makeCompletedFuture here
}
```

and because of the `@inlinable` we get the same perf and will have a nicer API. WDYT @Lukasa / @glbrntt ?

Reply:
SGTM

Reply:
Don't even need to do that I think, `makeCompletedFuture` does it for us :)